src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java

```diff
@@ -32,6 +35,25 @@ public OverseasEntitySubmissionDtoValidator(EntityNameValidator entityNameValida
 
     public Errors validateFull(OverseasEntitySubmissionDto overseasEntitySubmissionDto, Errors errors, String loggingContext) {
 
+        if (StringUtils.isNotBlank(overseasEntitySubmissionDto.getEntityNumber()) && isRoeUpdateEnabled) {
```

Comment:
You should check isRoeUpdateEnabled first.
if (sRoeUpdateEnabled && StringUtils.isNotBlank(overseasEntitySubmissionDto.getEntityNumber())